### DISCUSSION TITLE
Will Stride Natively Support Bepu Physics

### DISCUSSION THREAD

**Discussion by IndieDev99** [stride3d/stride#1858] [General]
Bepu Physics is very fast and robust Physics Engine. There is a test on YouTube by a Stride user. With Bepu he spawn upto 20,000 rigidbodies. With stride physics engine you can spawn upto 1200 to 1500 rigidbodies (Before physics update these were 600 now 3X more). So, will devs care about Bepu Physics engine (written completely in C#).

**Comment by tebjan** [comment]
Yes, the maintainers of Stride are undoubtedly interested in integrating the Bepu Physics engine due to its significant performance advantages. However, this integration is a substantial task. Stride is not backed by a company and relies on developers who contribute in their spare time.

The user who made the Bepu Physics test has to take responsibility of the integration themselves. It is highly likely that they will start the task, and the maintainers will offer support and guidance during the integration process. The task is likely to be more manageable since there is already an almost complete Stride integration of [Bepu in the Focus fork of Stride](https://github.com/phr00t/FocusEngine/tree/master/sources/engine/Xenko.Physics/Bepu). Ping @Nicogo1705.

It's worth mentioning that achieving high performance is also possible with Bullet Physics by leveraging GPU instancing.

So, while the integration of Bepu Physics into Stride is a promising prospect, it relies on community contributions and may take some time due to the nature of Stride's development team.

**Comment by IXLLEGACYIXL** [comment]
The video with 1200-1500 Physicsbodies was rigged and allready debunked ( i assume you talk about that one )

Bullet isnt slow. It never was.
Bullet has 1 little problem in C# area: none updates it 
https://github.com/AndresTraks/BulletSharp

Compared to Bepu which is still updated

**Reply by IndieDev99** [reply]
I know Bullet is very much mature but Bipu is actually something we care about.

**Comment by NicusorN5** [comment]
Since it is a major task, I think It would help having adding BEPUPhysics 2 support as a Open Collective project.

**Reply by VaclavElias** [reply]
If it is in the Open Collective we could ping Bepu Physics 2 devs if they are interested :)

**Reply by amerkoleci** [reply]
In case you're intersted I'm maintaining C# bindings for JoltPhysics, with C layer written manually for best performance https://github.com/amerkoleci/JoltPhysicsSharp

Feel free to ping me if you want to use it and require any help!

**Reply by IXLLEGACYIXL** [reply]
Bepus advantage is that its written in C# which is an aim for Stride to have only C# in the engine and get rid of everything that is C++ or atleast hide it behind a binding ( like Silk does it )

**Reply by amerkoleci** [reply]
Dude, isn't that the same I'm doing with the bindings? 😀 (Like Silk)

**Reply by IndieDev99** [reply]
> In case you're intersted I'm maintaining C# bindings for JoltPhysics, with C layer written manually for best performance https://github.com/amerkoleci/JoltPhysicsSharp
> 
> Feel free to ping me if you want to use it and require any help!

Thank you very much. We the ex Unity devs will be appreciating you for this.

**Reply by IndieDev99** [reply]
> Bepus advantage is that its written in C# which is an aim for Stride to have only C# in the engine and get rid of everything that is C++ or atleast hide it behind a binding ( like Silk does it )

You are right and Bepu is much faster than jolt physics

**Reply by sandsalamand** [reply]
Who is in charge of the Open Collective page? Let's add this.

**Reply by ArieLeo** [reply]
> > Bepus advantage is that its written in C# which is an aim for Stride to have only C# in the engine and get rid of everything that is C++ or atleast hide it behind a binding ( like Silk does it )
> 
> You are right and Bepu is much faster than jolt physics

Wait, seriously? 🤔I'm asking because i really don't know

**Reply by IXLLEGACYIXL** [reply]
dependent on the context

stride with bepu is faster than godot physics + jolt extension ( atleast in the made tests which dont cover everything ofcourse )

try it yourself 
https://github.com/Nicogo1705/BepuPhysicIntegrationTest

**Comment by VaclavElias** [comment]
Refernece [WIP], [Physics] Bepu integration https://github.com/stride3d/stride/pull/2131.
